backend/core/quivr_core/llm/llm_endpoint.py

```diff
@@ -21,13 +25,31 @@ def __init__(self, llm_config: LLMEndpointConfig, llm: BaseChatModel):
             self._config.model
         )
 
+        if llm_config.tokenizer_hub:
+            # To prevent the warning
+            # huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...
+            os.environ["TOKENIZERS_PARALLELISM"] = (
+                "false"
+                if not os.environ.get("TOKENIZERS_PARALLELISM")
+                else os.environ["TOKENIZERS_PARALLELISM"]
+            )
+            self.tokenizer = AutoTokenizer.from_pretrained(llm_config.tokenizer_hub)
+        else:
+            self.tokenizer = tiktoken.get_encoding("cl100k_base")
+
+    def count_tokens(self, text: str) -> int:
+        # Tokenize the input text and return the token count
+        encoding = self.tokenizer.encode(text)
+        return len(encoding)
+
     def get_config(self):
         return self._config
 
     @classmethod
     def from_config(cls, config: LLMEndpointConfig = LLMEndpointConfig()):
+        _llm: Union[AzureChatOpenAI, ChatOpenAI, ChatAnthropic]
         try:
-            if config.model.startswith("azure/"):
+            if config.supplier == DefaultLLMs.AZURE:
```

Comment:
To be decided... Ideally, we'd have a single interface for `_llm` (and a single `LLMEndpoint`) instead of `AzureChatOpenAI`, `ChatAnthropic`, `ChatOpenAI`, ... as they share most arguments, then pass supplier-specific options in a dict